java/vector/src/main/codegen/templates/DenseUnionVector.java

```diff
@@ -940,4 +940,13 @@ public void setInitialCapacity(int valueCount, double density) {
       }
     }
   }
+
+  /**
+   * Set the element at the given index to null. For DenseUnionVector, it is a no-op as nulls are not supported at the
+   * top level, and isNull() always returns false. Nullability is only handled at the level of individual child vectors.
```

Comment:
Hmm, should this raise an error then, instead of silently ignoring the request?

Reply:
It is a little confusing this way. An UnsupportedOperation exception seems to make sense, but this is the way it's handled in UnionWriter and DenseUnionWriter. I'm not sure we should change it to throw an exception now, but they both should work the same way.

Reply:
Oh, I see. Well, can you open a separate JIRA to discuss changing this?

Reply:
Is there a way to create unit test for this?

Reply:
@davisusanibar I could create a unit test, but since the method does nothing, it's hard to get it wrong.

Reply:
Could it also delegate to a child?

Also given this is a new method, I think an exception from the get-go is OK.

Reply:
@lidavidm 
> given this is a new method, I think an exception from the get-go is OK.
This _is_ a new method, but there is an existing, equivalent method in the FieldWriters called `writeNull()` which is implemented as a no-op. I don't think they can have different results. If this one throws an exception, that one should as well, which would be a breaking change. Is that your recommendation?

> Could it also delegate to a child?
I looked at delegating to a child vector, and, as with the above option, I think it needs to be consistent with the FieldWriter. Also, I think to ensure predictable results it would have to write null to _all_ the child vectors, which I wasn't sure was completely practical.

Reply:
Ok I will throw an exception.

Reply:
If it's possible to delegate we should (I still don't see the issue but haven't looked closely)

Reply:
I think UnsupportedOperation is the best solution (and would have been correct in FieldWriter).  I made that change to the code. 

There seem to be at least a couple potential issues with delegation from an API design standpoint. One is that if you only update one vector, you're effectively choosing the one at random (if you pick the first, it may already be null), and the user will get inconsistent results if they check for nullability in the child vectors. Also, you will have made a change to the data, but the naive test of the effects of the change (calling isNull()) will report the same result as before the change. 

> For dense union vectors, only one would need a value, and an offset would be appended.

This suggests a second issue. If there's only one child vector with a value before you setNull(), then none will have a value after you do. I'm not sure the result would be but it seems like it breaks the expectation.

Reply:
This is really more a design issue with union vectors, IMO. I would still prefer we delegate to the underlying vector but there's not a great choice here because of how unions work, there's always going to be some incongruity.

Reply:
In either case LGTM